### PR TITLE
[SVLS-5551] Add lambda threads enhanced metrics

### PR DESCRIPTION
**Pull request by shreyamalpani**
### What does this PR do?

This PR introduces two new enhanced lambda metrics. Each of the metrics are emitted once per invocation and represent thread usage by the lambda function.

The two new metrics are:

- `aws.lambda.enhanced.threads_max` - maximum number of threads that can be used by the function
- `aws.lambda.enhanced.threads_use` - maximum number of threads in use at once over the duration of the function

### Motivation

- Users want insight into thread resources used by their function
- [Product spec](https://docs.google.com/document/d/16i7VwCCbRgbhsX_gu8jtIOS6txPJmgb7X1sTHNiSjOM/edit?pli=1#heading=h.b3qf7mow8q5i)

### Additional Notes

* Considers the thread usage of all running processes

### Describe how to test/QA your changes

- [Build the lambda extension](https://datadoghq.atlassian.net/wiki/spaces/SLS/pages/2723611380/Lambda+Extension#Manual-Testing)
- Create a lambda function with the extension built in previous step
- Turn on lambda insights / add the [lambda insights extension](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Lambda-Insights-Getting-Started.html) to your function
- Invoke the function
- View the new enhanced metrics in metrics explorer or [this dashboard](https://ddserverless.datadoghq.com/dashboard/ker-x4u-kbr?fromUser=true&refresh_mode=sliding&view=spans&from_ts=1716303324812&to_ts=1716305124812&live=true)
- [Compare with lambda insight metrics](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Lambda-Insights-view-metrics.html) (lambda insight metric will be lower as it only accounts for the lambda process)

### CODE REVIEW

File: pkg/serverless/proc/testData/process/valid/31/task/3
Comment (TalUsvyatsky) [inline pkg/serverless/proc/testData/process/valid/31/task/3:1]: Is the purpose of this file to make sure we only count directories?

Comment (shreyamalpani) [reply pkg/serverless/proc/testData/process/valid/31/task/3:1]: Yup

**Review by TalUsvyatsky — APPROVED** [review]
🎉

**Pull request merged**